src/main/java/org/opensearch/ad/task/ADTaskManager.java

```diff
@@ -1514,6 +1520,7 @@ public void deleteADTasks(String detectorId, AnomalyDetectorFunction function, A
         client.execute(DeleteByQueryAction.INSTANCE, request, ActionListener.wrap(r -> {
             if (r.getBulkFailures() == null || r.getBulkFailures().size() == 0) {
                 logger.info("AD tasks deleted for detector {}", detectorId);
+                deleteADResultOfDetector(detectorId);
```

Comment:
I saw you call deleteADTasks inside DeleteAnomalyDetectorTransportAction. Do we wait until all results are deleted before returning delete success to users? There can be a huge amount of results.

Reply:
No, `deleteADResultOfDetector` just send out delete request, don't wait until it's done. After delete request sent out, will continue to delete AD job and return response to user.

Reply:
Ah I see now. 

I am hesitated on this approach because we can query/delete a huge number of docs.  we can have 10 shards each shard 65 GB for each rolled over index.  And the data is stored for 30 days.  We don't know how many result indices there.  Have you measured the performance impacts? The performance will be impacted depends on the index number/size and cluster resources/workload. Is there a max time before cancelling? There are throttling like mentioned in https://www.elastic.co/guide/en/elasticsearch/reference/current/docs-delete-by-query.html Do you consider using it?  

See the quote:
"However, this query is really slow. It scales linearly with document size. If you have enough documents that you need to be rotating them to prevent your Elasticsearch instance from bursting into flames, you probably can’t delete records this way, and will need to use time-based indices instead."

https://www.cloudsavvyit.com/7152/how-to-rotate-and-delete-old-elasticsearch-records-after-a-month/

Reply:
With historical analysis, user can generate huge amount of anomaly result by running a long period of time and even more by running HC detector. That may make AD result index roll over quickly, that will impact realtime AD result retention period. Will do some performance test later and tune this part. One option is we can add dynamic setting to let user control delete anomaly result or not as it's hard to say what's the user's most concern. If we have concern about performance based on the test result, we can set it as false by default.

Reply:
We changed a lot code recently, will redo performance test to tight up before release. Create one issue to track https://github.com/opensearch-project/anomaly-detection/issues/172

Reply:
I remember you created an API for users to delete results by themselves, right?  This can be a mitigation if issue arise.

A common solution for this problem is mark the stuff to delete and do that when the system is not busy. 

How about you disable this by default before your perf test indicate otherwise?

Check these posts, delete by query is not suitable for the use case.

https://discuss.elastic.co/t/searching-performance-dramatically-decreases-15s-if-the-indexation-process-is-enabled/37833/11

https://discuss.elastic.co/t/delete-by-query-performance-optimization/244437

https://discuss.elastic.co/t/alternative-for-delete-by-query-or-solution-for-performance-improvement/272298

Reply:
Thanks, as we discussed, for some smaller cluster which doesn't have big data volume, it's better to provide some auto way to help cleanup anomaly result. Agree that it's risky for big clusters. I will add dynamic setting for this and set default as close.